Applications/WebPortal/WebApp/library/source/pages/account_directory_page/local_account_directory_model.ts

```diff
@@ -0,0 +1,44 @@
+import * as Beam from 'beam';
+import { AccountDirectoryModel } from './account_directory_model';
+import { AccountEntry } from './account_entry';
+
+/** Implements an EntitlementsModel in memory. */
+export class LocalAccountDirectoryModel extends AccountDirectoryModel {
+
+  /** Constructs an new model.
+   * @param groupList - The account to represent.
+   * @param accountLists - The set of entitlements granted to the account.
+   */
+  constructor( groupList: Beam.Set<Beam.DirectoryEntry>,
+      accountLists: Beam.Map<Beam.DirectoryEntry, AccountEntry[]>) {
+    super();
+    this._isLoaded = false;
+    this._groupList = groupList;
+    this._accountLists = accountLists;
+  }
+
+  /** Returns true if this model has been loaded. */
+  public isLoaded(): boolean {
```

Comment:
This method is implemented in an overly hostile manner.

Reply:
Hostile?

Reply:
Nevermind. I figured it out.

Reply:
Yes, put yourself in the shoes of a user of this method, would you be satisfied with its behavior or would you want to take your monitor and throw it out of a window?